### ISSUE TITLE
Enhancement: [consistent-type-assertions] Suggest alternate styles for objectLiteralTypeAssertions

### ISSUE THREAD

**Issue by NotWoods** [typescript-eslint/typescript-eslint#6607]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-assertions/

### Description

With the `objectLiteralTypeAssertions` option turned on, we can suggest code changes to either swap to a type annotation or to use satisfies.

Since these both change TypeScript's behaviour, I thought it would be better to have them as suggestions instead of autofixes.

### Fail

```typescript
const x = { ... } as T;
```

### Pass

```typescript
const x: T = { ... }; // autofix for variable declarators
const x = { ... } satisfies T; // autofix for TypeScript 4.9+
```

### Additional Info

Happy to open a PR. I've previously implemented this in a Microsoft repo and would like to bring it upstream.

**Comment by bradzacher** [comment]
They'll have to be suggestion fixers because there's possibility of type errors from the change - but I think it would be great to do this!
